### PR TITLE
allow for playing of local files which are in the sonos music library

### PR DESCRIPTION
**Pull request by darioce**
This change allows to pass the argument _use_local_library_ and _sonos_ip_ to the skill. It will search for artists, albums or tracks in the local sonos library instead of spotify.

### CODE REVIEW

**Comment by alaa-saade** [comment]
Hi @darioce. Thanks for your PR and sorry for the delay in reviewing, we've been a bit taken lately. We'll try to take a look soon!

File: README.rst [hunk 1]
```diff
@@ -39,6 +40,23 @@ The skill allows you to control `Sonos <http://musicpartners.sonos.com/docs?q=no
...
 
+Useage with Sonos Local Music Library
+-------------------------------------
+
+If you would like to use your local music (e.g. on a NAS) you can specify this in the `Snipsfile <https://github.com/snipsco/snipsskills/wiki/The-Snipsfile>`_. The get_local_library_data.py allows you to pull your local artists, playlists etc. to include as custom slots in your assistant.
+
+.. code-block:: yaml
+
+    skills:
+      - url: https://github.com/darioce/snips-skills-sonos
```
Comment (akaisuisei) [inline README.rst]: please use this url:
https://github.com/snipsco/snips-skills-sonos

File: get_local_library_data.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+#! /usr/bin/env python
```
Comment (akaisuisei) [inline get_local_library_data.py]: I don't understand why you add a dumper script to the skill.

File: snipssonos/snipssonos.py [hunk 3]
```diff
@@ -125,77 +138,125 @@ def turn_on_radio(self, radio_name):
...
-            return None
-        self.device.stop()
-        self.device.clear_queue()
-        if _shuffle:
-            shuffle(tracks)
-        for track in tracks:
-            self.add_from_service(track['track']['uri'], self.spotify_service, True)
-        self.device.play_from_queue(0)
+        if self.local_library is True:
+            if self.my_library is None:
```
Comment (akaisuisei) [inline snipssonos/snipssonos.py]: Please refactor your code from line 142 to 151

File: snipssonos/snipssonos.py [hunk 4]
```diff
@@ -125,77 +138,125 @@ def turn_on_radio(self, radio_name):
...
-            return None
-        self.device.stop()
-        self.device.clear_queue()
-        for track in tracks:
-            self.add_from_service(track['uri'], self.spotify_service, True)
-        self.device.play_from_queue(0)
-
-    def play_album(self, album, _shuffle=False):
+        if self.local_library is True:
+            if self.my_library is None:
```
Comment (akaisuisei) [inline snipssonos/snipssonos.py]: Same here from line 171 to 179

File: snipssonos/snipssonos.py [hunk 5]
```diff
@@ -125,77 +138,125 @@ def turn_on_radio(self, radio_name):
...
-            return None
-        self.device.stop()
-        self.device.clear_queue()
-        if _shuffle:
-            shuffle(tracks)
-        for track in tracks:
-            self.add_from_service(track['uri'], self.spotify_service, True)
-        self.device.play_from_queue(0)
+        if self.local_library is True:
+            if self.my_library is None:
```
Comment (akaisuisei) [inline snipssonos/snipssonos.py]: same here from line 196 to 206

File: snipssonos/snipssonos.py [hunk 6]
```diff
@@ -125,77 +138,125 @@ def turn_on_radio(self, radio_name):
...
-            return
-        track = self.spotify.get_track(name)
-        if track is None:
-            return None
-        self.device.stop()
-        self.device.clear_queue()
-        self.add_from_service(track['uri'], self.spotify_service, True)
-        self.device.play_from_queue(0)
+        if self.local_library is True:
+            if self.my_library is None:
```
Comment (akaisuisei) [inline snipssonos/snipssonos.py]: same here from line 225 to 233

File: setup.py [hunk 7]
```diff
@@ -5,7 +5,7 @@
     version='1.0.0',
     description='Sonos skill for Snips',
     author='The Als',
-    url='https://github.com/snipsco/snips-skill-sonos',
+    url='https://github.com/darioce/snips-skill-sonos',
```
Comment (akaisuisei) [inline setup.py]: Please don't add this commit to your pull request

File: setup.py [hunk 8]
```diff
@@ -5,7 +5,7 @@
     version='1.0.0',
     description='Sonos skill for Snips',
     author='The Als',
-    url='https://github.com/darioce/snips-skill-sonos',
+    url='https://github.com/snipsco/snips-skill-sonos',
```
Comment (akaisuisei) [inline setup.py]: Please remove the commit to the pull request

**Comment by darioce** [comment]
@akaisuisei thank you a lot for cleaning up my messy PR! The current sonos-skill is a huge improvement for me and makes my Snips-Assistant much more useful!

**Pull request merged**